datastore/google/cloud/datastore/key.py

```diff
@@ -333,6 +338,15 @@ def from_legacy_urlsafe(cls, urlsafe):
         :rtype: :class:`~google.cloud.datastore.key.Key`.
         :returns: The key corresponding to ``urlsafe``.
         """
+        if not isinstance(urlsafe, basestring):
+            raise TypeError('urlsafe must be a string; received %r' % urlsafe)
+        if isinstance(urlsafe, unicode):
+            urlsafe = urlsafe.encode('utf8')
+
+        mod = len(urlsafe) % 4
+        if mod:
+            urlsafe += '=' * (4 - mod)
```

Comment:
1.  Please move this to **after** the `_to_bytes` call
1.  It'll be a bit annoying to have the `if` branch in there for unit tests, so instead I recommend

    ```python
    padding = b'=' * (-len(urlsafe) % 4)
    urlsafe += padding
    ```

----

To sanity check that last bit, observe:

```python
>>> def pad(urlsafe):
...     padding = b'=' * (-len(urlsafe) % 4)
...     urlsafe += padding
...     return urlsafe
... 
>>> for len_ in range(10):
...     value = ''.join(map(str, range(1, len_ + 1)))
...     print(pad(value))
... 

1===
12==
123=
1234
12345===
123456==
1234567=
12345678
123456789===
```